src/frontend/src/pages/HomePage/components/WorkPackagesSelectionView.tsx

```diff
@@ -0,0 +1,130 @@
+import { RoleEnum, wbsPipe, WorkPackage } from 'shared';
+import { Box, Grid, useTheme } from '@mui/material';
+import {
+  getInProgressWorkPackages,
+  getOverdueWorkPackages,
+  getUpcomingWorkPackages
+} from '../../../utils/work-package.utils';
+import { useCurrentUser } from '../../../hooks/users.hooks';
+import PageBlock from '../../../layouts/PageBlock';
+import WorkPackageCard from './WorkPackageCard';
+import WorkPackageSelect from './WorkPackageSelect';
+import React, { useState } from 'react';
+import EmptyPageBlockDisplay from './EmptyPageBlockDisplay';
+import CheckCircleOutlineOutlinedIcon from '@mui/icons-material/CheckCircleOutlineOutlined';
+
+const NoWorkPackages: React.FC = () => {
+  return (
+    <Box
+      sx={{
+        width: '100%',
+        height: '40vh',
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center'
+      }}
+    >
+      <EmptyPageBlockDisplay
+        icon={<CheckCircleOutlineOutlinedIcon sx={{ fontSize: 70 }} />}
+        heading={`You're all caught up!`}
+        message={'You have no pending work packages of this type!'}
+      />
+    </Box>
+  );
+};
+
+const WorkPackagesSelectionView: React.FC = () => {
+  const user = useCurrentUser();
+  const theme = useTheme();
+
+  const relevantWPs = user.teamsAsHead
+    ? user.teamsAsHead.map((team) => team.projects.map((project) => project.workPackages)).flat(2)
+    : [];
+
+  relevantWPs.concat(
+    user.teamsAsLead ? user.teamsAsLead.map((team) => team.projects.map((project) => project.workPackages)).flat(2) : []
+  );
+
+  const upcomingWPs: WorkPackage[] = getUpcomingWorkPackages(relevantWPs);
+  const inProgressWPs: WorkPackage[] = getInProgressWorkPackages(relevantWPs);
+  const overdueWPs: WorkPackage[] = getOverdueWorkPackages(relevantWPs);
+
+  const workPackages: [string, WorkPackage[]][] = [
+    [`Upcoming Work Packages (${upcomingWPs.length})`, upcomingWPs],
+    [`In Progress Work Packages (${inProgressWPs.length})`, inProgressWPs],
+    [`Overdue Work Packages (${overdueWPs.length})`, overdueWPs]
+  ];
+
+  let defaultFirstDisplay = 2;
+  if (workPackages[2][1].length === 0) {
+    defaultFirstDisplay = 1;
+    if (workPackages[1][1].length === 0) {
+      defaultFirstDisplay = 0;
+    }
+  }
+
+  const [currentDisplayedWPs, setCurrentDisplayedWPs] = useState<number>(defaultFirstDisplay);
+
+  const handleChange = (event: number) => {
+    setCurrentDisplayedWPs(event);
+  };
+
+  const getWorkPackages = (key: number): WorkPackage[] => {
```

Comment:
I feel like you could also just index into the array when you need it instead of using a helper